### PR TITLE
Disallow giving less points than the current active punishment has

### PR DESCRIPTION
**Pull request by marzeq**
**Please describe the changes this PR makes and why it should be merged**
Rodry ~~forced me~~ asked me to do this so here I am. Title is pretty self explanatory.

**In this PR:**
- Code changes were not tested

### CODE REVIEW

File: src/commands/Staff/punishment.ts [hunk 1]
```diff
@@ -665,7 +666,7 @@ const command: Command = {
 
 async function calculatePunishment(user: Discord.User, points: PunishmentPoints): Promise<Punishment> {
 	const activePunishments = await getActivePunishments(user),
-		hasActivePunishment = activePunishments.some(p => p.ended === false),
+		activePunishmentPoints = activePunishments.find(p => p.ended === false)?.points || null,
```
Comment (ImRodry) [inline src/commands/Staff/punishment.ts:669]: Although this doesn’t actually change anything, nullish coalescing is better
```suggestion
		activePunishmentPoints = activePunishments.find(p => p.ended === false)?.points ?? null,
```

File: src/commands/Staff/punishment.ts [hunk 2]
```diff
@@ -134,6 +134,7 @@ const command: Command = {
 			if (!checkPermissions(interaction.member as Discord.GuildMember, punishment)) throw "You don't have permission to give this punishment"
 			if (user.bot) throw "You cannot punish bots!"
 			if (member?.roles.cache.has("768435276191891456")) throw "You cannot punish this user!" //Discord Staff
+			if (points! < (punishment.activePunishmentPoints ?? 0)) throw "You cannot punish this user with less points than they have!"
```
Comment (ImRodry) [inline src/commands/Staff/punishment.ts:137]: We shouldn’t allow equal points either. Also feel free to tweak this message a little if you have better ideas but it should include the amount of points. 
```suggestion
			if (points! <= (punishment.activePunishmentPoints ?? 0)) throw `This user currently has an active punishment with ${punishment.activePunishmentPoints} points, so you cannot give them a less severe punishment!`
```

**Review by ImRodry — CHANGES_REQUESTED** [review]
Small changes, otherwise LGTM

File: src/commands/Staff/punishment.ts (hunk 2 above)
Comment (marzeq) [reply src/commands/Staff/punishment.ts:137]: The message is ok.

**Review by ImRodry — APPROVED** [review]

**Pull request merged**